dkist/dataset/tiled_dataset.py

```diff
@@ -150,6 +128,18 @@ def __init__(
         self._data = np.ma.masked_array(dataset_array, dtype=object, mask=mask)
         meta = meta or {}
         inventory = meta.get("inventory", inventory or {})
+
+        # If headers are saved as one Table for the whole TiledDataset, use those first
+        # Otherwise stack the headers saved for component Datasets
+        if meta.get("headers") is None:
+            ds_headers = [Table(ds.headers) if ds else Table({}) for ds in self._data.compressed()]
```

Comment:
Should we be using `if ds` here?

When is ds falsey? isn't .compressed() removing anywhere where the mask is True? So why would we have mask False and not a valid ds?